tests/autotest.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+BRANCH_TO_TEST=$(git branch | sed -n -e 's/^\* \(.*\)/\1/p')
+RESULT_DIR="/tmp/libmultilabel-test-$(whoami)/runs"
+REPORT_PATH="test_report.txt"
+
+#######################################
+# Run the command and get the test results in the latest log file.
+# Arguments:
+#   $1: Path prefix of the log file.
+#   $2: Command to run.
+#######################################
+get_test_results() {
+  prefix=$1
+  command="$2"
+
+  echo "Running $command ..."
+  $command
+
+  log_file=$(ls -t $prefix*[0-9]*/logs.json | head -1)
+  echo "Getting test results from $log_file ..."
+  while IFS="=" read -r key value; do
+      results[$key]=$value
+  done < <(jq -r '.test[0]|to_entries|map("\(.key)=\(.value)")|.[]' $log_file)
+}
+
+#######################################
+# Test results between current branch and master branch.
+# Arguments:
+#   $1: Data name such as MIMIC-50 or rcv1.
+#   $2: Nework name defined in libmultilabel/nn/networks/*
+#   $3: Command template to run.
+#######################################
+run_test() {
+  data_name=$1
+  network_name=$2
+  command=$(echo "$3" | sed "s/%s/$data_name/" | sed "s/%s/$network_name/")
+
+  declare -A results actual_results
+  prefix="${RESULT_DIR}/${data_name}_${network_name}_"
+
+  # Get actual results in current branch
+  get_test_results $prefix "$command"
+  for i in "${!results[@]}"; do actual_results[$i]=${results[$i]}; done
+
+  # Get expected results in master branch
+  git checkout master
```

Comment:
So we assume that users run this .sh in the main directory of LibMultiLabel.. But because this file is in tests/, users may directly call it in test/
Can do python3 ../main.py --config ../exa... in line 66???

Reply:
Oh yes .. That would be a problem. Maybe we can terminate the program if we find that they are in the `tests/` dir? (like the one we have in the Suggested change L80)

Because if user run this script under `tests/`, they would need to modify the relative paths in the configuration files like  `../data/MIMIC-50/train.txt`.